tools/strobelight/cli_function_profiler.py

```diff
@@ -0,0 +1,307 @@
+import functools
+import logging
+import os
+import re
+import subprocess
+import time
+from threading import Lock
+from typing import Any, List, Optional, Sequence
+
+# mypy: disallow-untyped-defs
```

Comment:
I think the convention is usually these go at the very top of the file since they apply to the whole file.